code/mob/living/critter/small_animal.dm

```diff
@@ -447,22 +459,113 @@ ABSTRACT_TYPE(/mob/living/critter/small_animal)
+				H.emote("sneeze")
+		return
+
+	seek_target(range)
+		. = list()
+		if (src.recent_pug_pet)
+			for (var/mob/living/carbon/human/H in hearers(range, src))
+				if (ispug(H) && isalive(H))
+					. += H
+					return
+		for (var/mob/living/critter/small_animal/mouse/C in hearers(range, src))
+			if (isdead(C)) continue
+			. += C
+		for (var/obj/critter/livingtail/C in hearers(range, src))
+			. += C
+
+		if (length(.) && prob(20))
+			playsound(src.loc, 'sound/voice/animal/cat_hiss.ogg', 50, 1)
+			src.visible_message("<span class='alert'>[src] hisses!</span>")
+
+	critter_attack(var/the_target)
+		if (istype(the_target, /obj/critter))
+			var/obj/critter/C = the_target
+			if (C.health <= 0 && C.alive)
+				playsound(src.loc, 'sound/impact_sounds/Generic_Hit_1.ogg', 50, 1, -1)
+				C.health -= 2
+		else if (istype(the_target, /mob))
+			var/mob/target = the_target
+			if(istype(target, /mob/living/critter/small_animal/mouse/weak/mentor) && prob(90))
+				src.visible_message("<span class='combat'><B>[src]</B> tries to bite [target] but \the [target] dodges [pick("nimbly", "effortlessly", "gracefully")]!</span>")
+				return
+			if ((src.catnip || prob(2) ) && (last_claw_fury < (TIME + 15 SECONDS)))
+				var/attackCount = rand(5, 9)
+				var/iteration = 0
+				while (iteration <= attackCount && (get_dist(src, target) <= 1))
+					if (iteration == 0)
+						target.setStatus("weakened", 2 SECONDS)
+						src.visible_message("<span class='combat'>[src] [pick("starts to claw the living <b>shit</b> out of ", "unleashes a flury of claw at ")] [target]!</span>")
+					random_brute_damage(target, src.attack_damage + 2, 1)
+					playsound(src.loc, 'sound/impact_sounds/Flesh_Tear_3.ogg', 60, 1)
+					iteration ++
+					sleep(0.3 SECONDS)
+			else
+				var/datum/targetable/critter/pounce/pounce = src.abilityHolder.getAbility(/datum/targetable/critter/pounce)
+				if (!pounce.disabled && pounce.cooldowncheck() && prob(50))
+					src.visible_message("<span class='combat'><B>[src]</B> pounces on [target] and trips them!</span>", "<span class='combat'>You pounce on [target]!</span>")
+					pounce.handleCast(target)
+				else
+					if (prob(50))
+						src.visible_message("<span class='combat'><B>[src]</B> pounces on [target]!</span>", "<span class='combat'>You pounce on [target]!</span>")
+						playsound(src.loc, 'sound/impact_sounds/Generic_Hit_1.ogg', 50, 1, -1)
+						random_brute_damage(target, src.attack_damage, 1)
+					else
+						src.visible_message("<span class='combat'><B>[src]</B> scratches [target]!</span>", "<span class='combat'>You scratch on [target]!</span>")
+						playsound(src.loc, 'sound/impact_sounds/Flesh_Tear_3.ogg', 50, 1, -1)
+						random_brute_damage(target, src.attack_damage + 2, 1)
+						if (prob(10))
+							bleed(target, 2)
+							boutput(target, "<span class='alert'>[src] scratches you hard enough to draw some blood! [pick("Bad kitty", "Piece of shit", "Ow")]!</span>")
```

Comment:
this proc would be much more readable if it used early returns instead of else chains